### ISSUE TITLE
Inconsistencies and questions on last release

### ISSUE THREAD

**Issue by fosterlynn** [valueflows/valueflows#580]
This is the result of a careful review done by yeff in the holochain forum, very helpful!  I'm working on them and will do PRs in valueflows, vf-graphql, and vf-json-schema.  (Some of it is inconsistencies in the different views, which were done by hand and sometimes by different people.  And there are a few that may require some discussion.)

[valueflows-remarks-v2(1).pdf](https://github.com/valueflows/valueflows/files/3622033/valueflows-remarks-v2.1.pdf)

**Comment by fosterlynn** [comment]
These are fixed, to the best of my ability, PRs done.  The vf-graphql PR is not yet merged, the other two are.

Some responses where there was not a fix:

>The “name” property hasn’t got a domain in the ontology. Is this correct?
Idem for “image”, “uri”, ...

We left these looser in the rdf defined vocabulary than in the json-schema or the graphql references.  Partly this is the way the vocab developed historically, and perhaps they should be tightened up in the rdf definitions?  Partly it is the looser structural nature of rdf.

>In the JSON-schema, the “inScopeOf” property points to Agent.json.  Is this correct?

I tightened this up in the json-schemas, mostly because I don't know how to make it more flexible.  I think that almost all of the inScopOf references will be to group agents, but I think we need to leave it flexible for now where possible, and see how it works out in practice with user groups.

>In the JSON-schema the “agreedIn” points to Agreement.json, while in the UML it has no
association.

I changed the json-schemas.  The agreedIn property as of now is a reference to something not in VF, could be any kind of agreement.  The Agreement in VF is more specific to a grouping of Commitments that have been agreed on by 2 or more Agents.

>I have a remark about the “hasBeginning” and “hasEnd” properties, present in
Commitment and many others. In the ontology they are defined, but I have a hard time
finding how they are connected with the classes. Maybe I don’t know where to look, but I
can’t find them being used in the ontology by the classes.

These sort of link into the time ontology, where we are using several properties, see this page https://valueflo.ws/specification/external-terms.html.  The way hasBeginning and hasEnd are defined in the rdf is magic done by @elf-pavlik , which he will have to explain, beyond my understanding.

>The UML and XML-schema have the property “hasDuration”.  I can’t find anything in the ontology about this property. Where should I look?

>The JSON-schema and UML have the “before” property. I can’t find it in the ontology.
(Also check the “after” property...

Both also here https://valueflo.ws/specification/external-terms.html.  Looks like we need to make that more clear in the doc.
